src/rust/engine/src/lib.rs

```diff
@@ -313,11 +313,34 @@ pub extern "C" fn scheduler_metrics(
 ) -> Handle {
   with_scheduler(scheduler_ptr, |scheduler| {
     with_session(session_ptr, |session| {
-      let values = scheduler
+      let mut values = scheduler
         .metrics(session)
         .into_iter()
         .flat_map(|(metric, value)| vec![externs::store_utf8(metric), externs::store_i64(value)])
         .collect::<Vec<_>>();
+      let values_ref = &mut values;
+      if session.should_record_zipkin_spans() {
+        let workunits = &session
```

Comment:
This `&` shouldn't be necessary:
```suggestion
        let workunits = session
```